control/grpc.py

```diff
@@ -70,12 +70,14 @@ def __init__(self):
         self.subsys_dhchap_key = defaultdict(dict)
         self.host_dhchap_key = defaultdict(dict)
         self.host_psk_key = defaultdict(dict)
+        self.host_nqn = defaultdict(dict)
```

Comment:
```suggestion
        self.host_nqn = defaultdict(set)
```

It seems your intention is to group "hosts" by the subsystem "nqn", with each subsystem "nqn" storing a unique collection of "host" values (no duplicates). Below is the logic for adding, removing, and counting items.

Reply:
OK